src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/events/PluginPatchsetCreatedEvent.java

```diff
@@ -225,6 +225,14 @@ public boolean shouldTriggerOn(GerritTriggeredEvent event) {
         if (excludeDrafts && ((PatchsetCreated)event).getPatchSet().isDraft()) {
             return false;
         }
+        if (StringUtils.isNotEmpty(commitMessageContainsRegEx)) {
+            if (commitMessagePattern == null) {
+                commitMessagePattern = Pattern.compile(
+                        this.commitMessageContainsRegEx, Pattern.DOTALL | Pattern.MULTILINE);
+            }
+            String commitMessage = ((PatchsetCreated)event).getChange().getCommitMessage();
+            return commitMessagePattern.matcher(commitMessage).find();
```

Comment:
Returning this early will will nullify all the other exclude checks below

Reply:
True. I didn't think of that.  

In my opinion, the `ManualPatchsetCreated` event should behave just like a `PatchsetCreated` event.  That is, the two triggers should be identical.  it's just that one allow you to retrigger jobs without having to actually create a new Gerrit patch set.

If that is the case, can't we just remove the `event instanced ManualPatchsetCreated` block?